src/v2/Components/FlashBanner/__tests__/FlashBanner.jest.tsx

```diff
@@ -1,32 +1,214 @@
+import { graphql } from "react-relay"
+import { Banner } from "@artsy/palette"
+import { flushPromiseQueue, renderRelayTree } from "v2/DevTools"
+import { mount } from "enzyme"
+
+jest.unmock("react-tracking")
+jest.unmock("react-relay")
 
 const { location: originalLocation } = window
 
+const getRelayWrapper = async ({
+  data,
+  mutationResults = {},
+  queryString: search = "",
+  props: passedProps = {},
+}) => {
+  delete window.location
+  window.location = { search } as any
+
+  return await renderRelayTree({
+    Component: props => {
+      return <FlashBanner {...props} {...passedProps} />
+    },
+    query: graphql`
+      query FlashBannerTestQuery @raw_response_type {
+        me {
+          canRequestEmailConfirmation
+        }
+      }
+    `,
+    variables: {},
+    mockData: data,
+    mockMutationResults: mutationResults,
+  })
+}
+
+afterEach(() => {
+  window.location = originalLocation
+})
+
 describe("FlashBanner", () => {
-  afterEach(() => {
-    window.location = originalLocation
+  it("renders nothing if no banner applies", async () => {
+    const wrapper = mount(<FlashBanner />)
+
+    expect(wrapper.find(Banner).exists()).toBeFalsy()
   })
 
-  it("renders a banner based on a messageCode prop", () => {
-    const wrapper = mount(<FlashBanner messageCode="confirmed" />)
-    expect(wrapper.text()).toContain("Your email has been confirmed.")
+  it("renders based on a contentCode prop", async () => {
+    const wrapper = mount(<FlashBanner contentCode="already_confirmed" />)
+
+    expect(wrapper.text()).toContain("You have already confirmed your email")
   })
 
-  it("returns nothing without complaint if the message code is not supported", () => {
-    const wrapper = mount(<FlashBanner messageCode="bad_message" />)
-    expect(wrapper.html()).toBeNull()
+  it("renders based on props.me.canRequestEmailConfirmation (from relay)", async () => {
+    const wrapper = await getRelayWrapper({
+      data: {
+        me: { id: "woot", canRequestEmailConfirmation: true },
+      },
+    })
+
+    expect(wrapper.text()).toContain("Please verify your email address")
   })
 
-  it("can determine a messageCode from the query string", () => {
-    delete window.location
-    const fakeLocation: any = { search: "?flash_message=expired_token" }
-    window.location = fakeLocation
+  it("renders based on a flash_message query param", async () => {
+    window.location = { search: "?flash_message=expired_token" } as any
 
     const wrapper = mount(<FlashBanner />)
 
     expect(wrapper.text()).toContain("Link expired. Resend verification email.")
   })
+
+  it("returns nothing for an unsupported content code", async () => {
+    const wrapper = mount(<FlashBanner contentCode="porntipsguzzardo" />)
+
+    expect(wrapper.find(Banner).exists()).toBeFalsy()
+  })
+
+  it("contentCode takes precedence over query string and props.me", () => {
+    window.location = { search: "?flash_message=expired_token" } as any
```

Comment:
loving this `delete window.location` trick 👌

Reply:
confused here because i'm no longer deleting window.location?

Reply:
I imagine Chris is getting at that `delete window.location` above enables the test setup at this line -- but clearly I'm not him and could be wrong.

Reply:
^